src/TriangleNodesProvisioner.cpp

```diff
@@ -105,6 +105,61 @@ namespace blitzdg {
         }
     }
 
+    void TriangleNodesProvisioner::computeGradVandermondeMatrix(index_type N,  const real_vector_type & r, const real_vector_type & s, real_matrix_type & V2Dr, real_matrix_type & V2Ds) const {
+        const index_type Nr = r.length(0);
+        const index_type Np = (N+1)*(N+2)/2;
+
+        real_vector_type a(Nr), b(Nr);
+
+        rsToab(r, s, a, b);
+
+        index_type count = 0;
+        for (index_type i=0; i < N+1; ++i) {
+            for (index_type j=0; j < N-i+1; ++j) {
+
+                real_vector_type v2drCol(Np), v2dsCol(Np);
+                evaluateGradSimplex(a, b, i, j, v2drCol, v2dsCol);
+
+                V2Dr(Range::all(), count) = v2drCol;
+                V2Ds(Range::all(), count) = v2dsCol;
+                ++count;
+            }
+        }
+    }
+
+    void TriangleNodesProvisioner::computeDifferentiationMatrices(const real_matrix_type & V2Dr, const real_matrix_type & V2Ds, const real_matrix_type & V, real_matrix_type & Dr, real_matrix_type & Ds) const {
+        firstIndex ii;
+        secondIndex jj;
+ 
+        const int numRowsV = V.rows();
+        const int numColsV = V.cols();
+
+
+		// Note: this is not a column major ordering trick. We need these transposes.
+        real_matrix_type Vtrans(numColsV, numRowsV);
+        real_matrix_type V2Drtrans(numColsV, numRowsV);
+        real_matrix_type V2Dstrans(numColsV, numRowsV);
+
+        real_matrix_type Drtrans(numColsV, numRowsV);
+        real_matrix_type Dstrans(numColsV, numRowsV);
+
+        Vtrans = V(jj, ii);
+        V2Drtrans = V2Dr(jj, ii);
+        V2Dstrans = V2Ds(jj, ii);
+
+
+        // Dr = V2Dr / V;
```

Comment:
I might write this comment as Dr = V2Dr * V^{-1}, since this Matlab syntax is nonstandard. Same for the comment below.